app/grants/management/commands/round8_payouts.py

```diff
@@ -0,0 +1,433 @@
+                    w3 = get_web3(network)
+                    tx_args = {
+                        'nonce': w3.eth.getTransactionCount(from_address),
+                        'gas': 8000000, # Set gas limit to 8M so we have even more margin
+                        'gasPrice': int(float(recommend_min_gas_price_to_confirm_in_time(1)) * 10**9 * 1.4)
+                    }
+                    
+                    tx = match_payouts.functions.setPayouts(payout_mapping).buildTransaction(tx_args)
+                    signed = w3.eth.account.signTransaction(tx, from_pk)
+                    
+                    tx_id = None
+                    success = False
+                    counter = 0
+                    while not success:
+                        try:
+                            tx_id = w3.eth.sendRawTransaction(signed.rawTransaction).hex()
+                            success = True
+                        except Exception as e:
+                            counter +=1
+                            if 'replacement transaction underpriced' in str(e):
+                                print(f'replacement transaction underpriced. retrying {counter}')
+                                time.sleep(WAIT_TIME_BETWEEN_TXS)
+
+                                # rebuild txn
+                                tx_args['gasPrice'] = int(float(recommend_min_gas_price_to_confirm_in_time(1)) * 10**9 * 1.6) # bump up multiplier to 1.6
+                                tx = match_payouts.functions.setPayouts(payout_mapping).buildTransaction(tx_args)
+                                signed = w3.eth.account.signTransaction(tx, from_pk)
+                            elif 'nonce too low' in str(e):
+                                print(f'nonce too low. retrying {counter}')
+                                time.sleep(WAIT_TIME_BETWEEN_TXS)
+
+                                # rebuild txn
+                                tx_args['nonce'] = w3.eth.getTransactionCount(from_address)
+                                tx = match_payouts.functions.setPayouts(payout_mapping).buildTransaction(tx_args)
+                                signed = w3.eth.account.signTransaction(tx, from_pk)
+                            else:
+                                raise e
+
+                    if not tx_id:
+                        raise Exception('Cannot set mapping')
+
+                    # Mapping has been successfully set
+                    # wait for tx to clear
+                    while not has_tx_mined(tx_id, network):
+                        time.sleep(1)
+
+                    # Pause until the next one
+                    print("SLEEPING")
+                    time.sleep(WAIT_TIME_BETWEEN_TXS)
+                    print("DONE SLEEPING")
+                
+                # All payouts have been successfully set, so now we update the database
+                for match in unpaid_scheduled_matches.order_by('amount'):
```

Comment:
wait why are these generated in the script? wont they be generated when the user redeems the txn?

also i think having the object creation in a helper function would be easier to manage

Reply:
They're generated here because that's how the old script worked and I wasn't sure what the purpose of these objects is 😛

We can generate them when a user redeems the transaction, but the only hurdle is there's no guarantee they initiate the claim from Gitcoin's site. They could do it by interacting with the contract directly, or a benevolent community member might write a script that sends that claim transaction on behalf of everyone (which would be easy to do). So the "right" way to generate these on claim is to monitor the contract events for claims and create objects then—does Gitcoin have infra setup for something like that?